lib/matplotlib/testing/conversion_cache.py

```diff
@@ -0,0 +1,200 @@
+        The flush method will delete files until their combined size is
+        under this limit, in bytes. Defaults to 100 megabytes.
+
+    """
+
+    def __init__(self, directory=None, max_size=int(1e8)):
+        self.gets = set()
+        self.hits = set()
+        if directory is not None:
+            self.cachedir = directory
+        else:
+            self.cachedir = self.get_cache_dir()
+        self.ensure_cache_dir()
+        if not isinstance(max_size, int):
+            raise ValueError("max_size is %s, expected int" % type(max_size))
+        self.max_size = max_size
+        self.cached_ext = '.png'
+        self.converter_version = {}
+        try:
+            self.converter_version['.pdf'] = \
+                checkdep_ghostscript()[1].encode('utf-8')
+        except:
+            pass
+        try:
+            self.converter_version['.svg'] = \
+                checkdep_inkscape().encode('utf-8')
+        except:
+            pass
+        self.hash_cache = {}
+
+    def get(self, filename, newname):
+        """Query the cache.
+
+        Parameters
+        ----------
+        filename : str
+            Full path to the original file.
+        newname : str
+            Path to which the result should be written.
+
+        Returns
+        -------
+        bool
+            True if the file was found in the cache and is now written
+            to `newname`.
+        """
+        self.gets.add(filename)
+        hash_value = self._get_file_hash(filename)
+        cached_file = os.path.join(self.cachedir, hash_value + self.cached_ext)
+        with cbook.Locked(self.cachedir):
+            if os.path.exists(cached_file):
+                shutil.copyfile(cached_file, newname)
+                self.hits.add(filename)
+                return True
+            else:
+                return False
+
+    def put(self, original, converted):
+        """Insert a file into the cache.
+
+        Parameters
+        ----------
+        original : str
+            Full path to the original file.
+        converted : str
+            Full path to the png file converted from the original.
+        """
+        hash_value = self._get_file_hash(original)
+        cached_file = os.path.join(self.cachedir, hash_value + self.cached_ext)
+        with cbook.Locked(self.cachedir):
+            shutil.copyfile(converted, cached_file)
+
+    def _get_file_hash(self, path, block_size=2 ** 20):
+        if path in self.hash_cache:
+            return self.hash_cache[path]
+        md5 = hashlib.md5()
+        with open(path, 'rb') as fd:
+            while True:
+                data = fd.read(block_size)
+                if not data:
+                    break
+                md5.update(data)
+        _, ext = os.path.splitext(path)
+        version_tag = self.converter_version.get(ext)
+        if version_tag:
+            md5.update(version_tag)
+        else:
+            warnings.warn(("Don't know the external converter for %s, cannot "
```

Comment:
As a user, if I see this message, I assume the "solution" is to delete the cache directory?